### ISSUE TITLE
History command with prefix/contains does not handle "--" to stop further option parsing

### ISSUE THREAD

**Issue by rocketraman** [fish-shell/fish-shell#2946]
If I want to delete a command starting with "--" from my history, I would normally do:

`history --delete --prefix -- "---"`

where the `--` stops further parsing of options. However, the history delete does not work when this is done. It does not find any entries to delete.

If "tricking" history to continue by doing `history --delete --prefix "-"` the interactive prompt now finds the entry and lists it, but when selected for deletion, the deletion fails because the interactive prompt does not use a `--` on the underlying history command:

```
[0] cancel
[1] all

[2] --- some entry here
Delete which entries? > 2
history: Unknown option “--- some entry here” 
```

Doing `history --delete -- "--- some entry here"` does work.

---
Fish version: fish, version 2.2.0

Operating system: Fedora 23, rpm packages

Terminal or terminal emulator: Konsole

**Comment by faho** [comment]
2.3.0 supports "--" to delimit the main arguments from options. It doesn't do this for "--prefix", but that's because that can only take exactly one argument and whatever comes after it is taken to be that, so `history --prefix --delete` will search for "--delete".

In other words this is fixed by a7ecc0db1fd396ca00431c60c53b593660110514, which will be in 2.3.0. Thanks!

**Comment by rocketraman** [comment]
Thank you!
